### PR TITLE
Treat warnings as errors

### PR DESCRIPTION
**Pull request by adizaimi**
Treat warnings as errors

### CODE REVIEW

File: bdb/file.c [hunk 1]
```diff
@@ -7441,13 +7428,12 @@ int bdb_check_files_on_disk(bdb_state_type *bdb_state, const char *tblname,
...
         logmsg(LOGMSG_ERROR, "%s: tablename too long\n", __func__);
         *bdberr = BDBERR_MISC;
-        free(buf);
         closedir(dirp);
         return -1;
     }
 
     /* for each file in the db's directory */
-    while ((error = readdir_r(dirp, buf, &ent)) == 0 && ent != NULL) {
+    while ((ent = readdir(dirp)) != 0) {
```
Comment (riverszhang89) [inline bdb/file.c]: bb_readdir

Comment (adizaimi) [reply bdb/file.c]: thanks Rivers. Fixed.

File: berkdb/db/db_meta.c [hunk 2]
```diff
@@ -620,10 +620,10 @@ __db_new_original(dbc, type, pagepp)
 			if (gbl_core_on_sparse_file) {
 				char cmd[100];
 
-				snprintf(cmd, sizeof(cmd), "gcore %d",
-				    getpid());
+				snprintf(cmd, sizeof(cmd), "gcore %d", getpid());
 				printf("%s\n", cmd);
-				system(cmd);
+				int lrc = system(cmd);
```
Comment (riverszhang89) [inline berkdb/db/db_meta.c]: Maybe just (void)system(cmd) ...

File: berkdb/env/env_recover.c [hunk 3]
```diff
@@ -1508,8 +1508,8 @@ __db_apprec(dbenv, max_lsn, trunclsn, update, flags)
 
 
 	/* Close all the db files that are open. */
-	if ((ret = __dbreg_close_files(dbenv)) != 0)
-		goto err;
+    if ((ret = __dbreg_close_files(dbenv)) != 0)
```
Comment (riverszhang89) [inline berkdb/env/env_recover.c]: tabs.

**Comment by riverszhang89** [comment]
/cbuild-compile-only to not overwhelm robomark. Please remove my comment when the PR is ready to review.

File: berkdb/db/db_meta.c (hunk 2 above)
Comment (adizaimi) [reply berkdb/db/db_meta.c]: i tried that and did not silence that warning.

Comment (adizaimi) [reply berkdb/db/db_meta.c]: Tried again, alas does not like (void):
/comdb2/tests/tools/nemesis.c:234:9: error: ignoring return value of 'system', declared with attribute warn_unused_result [-Werror=unused-result]
         (void)system(cmd);
         ^

**Comment by adizaimi** [comment]
Having one last run, with roborivers and robomark.

**Comment by riverszhang89** [comment]
If you don't mind I'd like to sit down and discuss this unused return value with you.

**Comment by adizaimi** [comment]
/cbuild-compile-only -- Compile but don't submit the branch to robomark.

**Comment by adizaimi** [comment]
/cbuild-compile-only -- Compile but don't submit the branch to robomark.

File: bdb/bdb_net.c [hunk 4]
```diff
@@ -220,15 +220,21 @@ char *print_addr(struct sockaddr_in *addr, char *buf)
     int rc = getnameinfo((struct sockaddr *)addr, len, name, sizeof(name),
                          service, sizeof(service), 0);
     if (rc) {
-        strerror_r(rc, errbuf, sizeof(errbuf));
+        char *tmp = strerror_r(rc, errbuf, sizeof(errbuf));
```
Comment (riverszhang89) [inline bdb/bdb_net.c]: These are incorrect. First of all `char *strerror_r(int errnum, char *buf, size_t buflen);` is a GNU extension. Second of all, the GNU extension returns an error string, which means `tmp` will always be non-NULL.

File: bdb/file.c [hunk 5]
```diff
@@ -1972,7 +1972,10 @@ static void panic_func(DB_ENV *dbenv, int errval)
 
     pid = getpid();
     snprintf(buf, sizeof(buf), "pstack %d", pid);
-    system(buf);
+    int lrc = system(buf);
+    if (lrc == -1) {
```
Comment (riverszhang89) [inline bdb/file.c]: lrc != 0

File: berkdb/rep/rep_method.c [hunk 6]
```diff
@@ -290,9 +290,10 @@ __rep_start(dbenv, dbt, gen, flags)
 				logmsg(LOGMSG_FATAL, "%s:%d:%s: Exiting after waiting too long for replication message thread.\n",
 					__FILE__, __LINE__, __func__);
 				pid = getpid();
-				snprintf(cmd, sizeof(cmd), "pstack %d",
-					(int)pid);
-				system(cmd);
+				snprintf(cmd, sizeof(cmd), "pstack %d", (int)pid);
+				int rc = system(cmd);
+				if (rc == -1)
```
Comment (riverszhang89) [inline berkdb/rep/rep_method.c]: rc != 0

**Comment by adizaimi** [comment]
Addressed all discussed issues. A previous version of this passed robomark:
cdb2test   Dec 28 15:19:20 2018  in-comdb2db AZ_warnings_as_errors.R20181228.5
and i will merge so we won't get any more warnings added by other checkins. 
If any issues arise i can fix with a later checkin.

**Comment by adizaimi** [comment]
Thank you Rivers and Akshat.

**Pull request merged**